code/obj/item/device/transfer_valve.dm

```diff
@@ -176,9 +187,10 @@
 				toggle = 1
 
 	process()
+		if(signalled)
+			UpdateIcon()
```

Comment:
This is really weird behavior to me - why not just  have a dirty icon var and set it to false at the end of icon updating

Reply:
Would you mind elaborating a bit on the first part of that?

Reply:
actually my idea wouldn't really work but why wouldn't you put the updateicon code below

Reply:
that's not what i was saying - why not just literally put the `UpdateIcon()` call in signal_start

Reply:
certain things (timer and prox sensor come to mind) change their sprite based on how much time is left or certain factors, so they'll need their sprites to be updated accordingly